data/src/test/java/com/teamagam/gimelgimel/data/message/repository/cache/room/dao/SearchMessagesDaoTest.java

```diff
@@ -44,48 +58,50 @@ public void tearDown() {
   @Test
   public void TextMessageSearchTest() {
     //Arrange
-    ChatMessageEntity chatMessage = mock(ChatMessageEntity.class);
-    chatMessage.text = "static and benel are the best!";
+    ChatMessage chatMessage =
+        new ChatMessage("1", "Dany", new Date(), new TextFeature("No on has ever call me danny"));
+    ChatMessage chatMessage2 =
+        new ChatMessage("2", "John", new Date(), new TextFeature("no Danny"));
+    ChatMessage chatMessage3 =
+        new ChatMessage("3", "John", new Date(), new TextFeature("How about my queen?"));
 
-    ChatMessageEntity chatMessage2 = mock(ChatMessageEntity.class);
-    chatMessage2.text = "new york is very cool place to live in.";
+    mMessagesDao.insertMessage(mMessagesEntityMapper.mapToEntity(chatMessage));
+    mMessagesDao.insertMessage(mMessagesEntityMapper.mapToEntity(chatMessage2));
+    mMessagesDao.insertMessage(mMessagesEntityMapper.mapToEntity(chatMessage3));
 
-    ChatMessageEntity chatMessage3 = mock(ChatMessageEntity.class);
-    chatMessage3.text = "I don't like new york";
-    mMessagesDao.insertMessage(chatMessage);
-    mMessagesDao.insertMessage(chatMessage2);
-    mMessagesDao.insertMessage(chatMessage3);
-
-    String searchText = makeStringReadyForSearch("new york");
+    String searchText = "new york";
 
     //Act
-    List<ChatMessageEntity> results = mSearchMessagesDao.getMessagesMatchingSearch(searchText);
+    List<ChatMessage> results = mMessagesTextSearcher.searchMessagesByText(searchText);
 
     //Assert
-    assertTrue(doesCollectionHasAllItems(results, chatMessage2, chatMessage3));
-  }
-
-  private String makeStringReadyForSearch(String text) {
-    text = "%" + text + "%";
-    return text;
+    assertThat(containsItems(results, chatMessage2, chatMessage3));
```

Comment:
That's not what I meant... I meant that you'd use a JUnit built-in matcher to assert that